### ISSUE TITLE
Could implement the rename function

### ISSUE THREAD

**Issue by mlambley** [gliterd/flysystem-backblaze#25]
https://github.com/gliterd/flysystem-backblaze/blob/42549be7b3e6f372c824896ccd1d901052cb6d8c/src/BackblazeAdapter.php#L130-L133

This could be implemented using copyFile (see https://github.com/gliterd/backblaze-b2/pull/57) and then delete the original file.

**Comment by remmelt** [comment]
Yes, a decision needs to be made what to do when the copy command succeeds, but the delete fails. There are no transactions ;-)

Perhaps throwing an exception is good enough.
